src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingSwitchDefaultCheck.java

```diff
@@ -35,10 +36,23 @@
  * cases are covered, this should be expressed in the
  * default branch, e.g. by using an assertion. This way
  * the code is protected against later changes, e.g.
- * introduction of new types in an enumeration type. Note that
- * the compiler requires switch expressions to be exhaustive,
- * so this check does not enforce default branches on
- * such expressions.
+ * introduction of new types in an enumeration type.
+ * </p>
+ * <p>
+ * This check does not validate any switch expressions. Rationale:
+ * the compiler requires switch expressions to be exhaustive. This means
+ * that all possible inputs must be covered.
+ * </p>
+ * <p>
+ * This check does not validate switch statements that use pattern or null
+ * labels. Rationale: switch statements that use pattern or null labels are
+ * checked by the compiler for exhaustiveness. This means that all possible
+ * inputs must be covered.
```

Comment:
> Why not combine these 2 paragraphs into 1? They both talk about it needing to be exhaustive and all inputs covered.

The idea is to make it very obvious that conditions for switch expressions and switch statements are different.

> To confirm, if all inputs are not covered this will result in a compile error, correct? Unless you disagree, maybe that point can also be added.

This is explained in comments in code below to make basic check description language as simple as possible.

Reply:
Admittedly, all requirements for compilation of switch statements are a bit complicated right now. This is probably due to Java encouraging exhaustiveness for **all** switch statements in the future.

From https://openjdk.java.net/jeps/420:

`Future compilers of the Java language may emit warnings for legacy switch statements that are not exhaustive.`

Reply:
It might be complicated wording will be if we combine. I tried to combine and it became hard to read.